site/ja/guide/create_op.md

```diff
@@ -0,0 +1,1231 @@
+# オペレーションを作成する
+
+Note: C++のカスタムオペレーションが、TensorFlow公式のpipパッケージとABI互換になることを保証するため、[Custom opリポジトリ](https://github.com/tensorflow/custom-op) のガイドにしたがってください。
+カスタムオペレーションをビルドして配布するためのDockerのイメージはもちろんのこと、端から端までコード例が示されています。
+
+既存のTensorFlowのライブラリに存在しないオペレーションを作りたい場合、既存のPythonのオペレーションや関数を組み合わせて、Pythonでオペレーションを書くことを推奨します。
+もしそれが不可能なら、C++のカスタムオペレーションを作ってもよいです。
+C++のカスタムオペレーションを作りたいと考える理由は、いくつかあります。
+
+* 既存のオペレーションの組み合わせでオペレーションを表現するのが、不可能または簡単ではない
+* 既存のプリミティブの組み合わせでオペレーションを表現するのが、効率的ではない
+* 将来コンパイラが融合することが難しいプリミティブの組み合わせを、自前で融合したい
+
+たとえば、"MaxPool" オペレーションと似ているが、最大値のかわりにウィンドウをスライドさせて中央値を計算する、"median pooling" のようなものを実装したいとしましょう。
+これは、オペレーションの組み合わせ（たとえば、ExtraImagePatchesとTopKを使う）でも可能ですが、1つの融合したオペレーションとしてより賢明に実装したネイティブなオペレーションと比較して、性能とメモリ効率の面で劣るかもしれません。
+オペレーションの組み合わせで、やりたいことを表現する試みには価値があります。
+もしそれが難しいまたは非効率であることが証明されたときのみ、新しいオペレーションを追加することを検討しましょう。
+
+カスタムオペレーションを組み込むために必要なことを、次に示します。
+
+1. C++ファイル内で新しいオペレーションを登録します。オペレーションの登録では、オペレーションの実装とは独立であるオペレーションの機能のためのインターフェース（仕様）を定義します。たとえば、オペレーションの登録では、オペレーション名やオペレーションの入出力を定義します。また、テンソルのシェイプ推論に使用されるシェイプ関数を定義します。
+2. C++でオペレーションを実装します。オペレーションの実装はカーネルとして知られ、Step 1で登録した仕様の実装を具体化します。異なる入出力型、アーキテクチャ（たとえば、CPUやGPU）のために複数のカーネルが存在することもあり得ます。
+3. Pythonのラッパーを作成する（任意）。このラッパーは、Pythonでオペレーションを作るときに使われるパブリックなAPIです。デフォルトのラッパーは、オペレーションの登録から生成され、直接利用することもできますし、追加することもできます。
+4. オペレーションの勾配を計算するための関数を書きます。（任意）
+5. オペレーションをテストします。便宜上、たいていはPythonで行いますが、C++でオペレーションをテストすることも可能です。勾配を定義した場合、Python からは `tf.test.compute_gradient_error` を使って確認できます。Reluのような順伝搬の関数とその勾配をテストするための例については、[`relu_op_test.py`](https://www.tensorflow.org/code/tensorflow/python/kernel_tests/relu_op_test.py) を見てください。
+
+
+## 前提条件
+
+* C++に精通していること
```

Comment:
[weak suggestion]
```suggestion
* C++になじみがあること
```
The current translation looks too strong.

Reply:
Good suggestion 👍